packages/assets-controllers/src/TokenRatesController.test.ts

```diff
@@ -295,7 +295,7 @@ describe('TokenRatesController', () => {
     expect(mock).not.toThrow();
   });
 
-  it('should update exchange rates when tokens change while polling is active', async () => {
+  it('should update exchange rates when tokens change', async () => {
```

Comment:
It looks like we have a test for when `allTokens` changes, but what about for when `allDetectedTokens` changes?

Reply:
We already have a test like this, though it's at the end of the test suite (the order doesn't make a great deal of sense; I think it's last because it was worked on last)

Reply:
Ah I see now.

In that case should we have a test that's the same as that test but verifies nothing happens when polling is disabled?

Reply:
Good catch. Since this is unrelated to the changes in this PR, I will address it in a follow-up PR to reorganize these tests and add few missing test cases

Reply:
Added in #1522